crates/core/src/suppress.rs

```diff
@@ -1,34 +1,25 @@
-use itertools::{EitherOrBoth, Itertools};
-use marzano_language::{
-    language::Language,
-    parent_traverse::{ParentTraverse, TreeSitterParentCursor},
-};
-use tree_sitter::{Node, Range};
-
 use crate::binding::Binding;
+use grit_util::AstNode;
+use itertools::{EitherOrBoth, Itertools};
+use marzano_language::language::Language;
+use marzano_util::node_with_source::NodeWithSource;
+use tree_sitter::Range;
 
 impl<'a> Binding<'a> {
     pub(crate) fn is_suppressed(&self, lang: &impl Language, current_name: Option<&str>) -> bool {
-        let (src, node) = match self {
+        let node = match self {
             Self::Node(src, node) | Self::List(src, node, _) | Self::Empty(src, node, _) => {
-                (src, node)
+                NodeWithSource::new(node.clone(), src)
             }
             Self::String(_, _) | Self::FileName(_) | Self::ConstantRef(_) => return false,
         };
-        let target_range = node.range();
-        for n in
-            node.children(&mut node.walk())
-                .chain(ParentTraverse::new(TreeSitterParentCursor::new(
-                    node.clone(),
-                )))
-        {
-            let mut cursor = n.walk();
-            let children = n.children(&mut cursor);
-            for c in children {
-                if !(lang.is_comment(c.kind_id()) || lang.is_comment_wrapper(&c)) {
+        let target_range = node.node.range();
+        for n in node.children().chain(node.ancestors()) {
```

Comment:
I've left the semantics intact, but to be honest, I still have my doubts about why we would want to chain children and ancestors together. The inner loop also loops over children, so why would we want to check grandchildren of a node for suppression comments?

Reply:
@seren5240 originally implemented this, do you recall why?

Since comment attachment points are somewhat hard to predict in tree-sitter, it does make sense for us to iterate up/down the tree a bit.

Reply:
I don't remember, but you can find out by removing it and seeing which unit tests fail.